packages/cloud_firestore/cloud_firestore/test/document_reference_test.dart

```diff
@@ -292,6 +295,27 @@ void main() {
           ),
         );
       });
+
+      test('can encode _WithConverterDocumentReference', () {
+        MethodChannelFirebaseFirestore.channel = const MethodChannel(
+          'plugins.flutter.io/firebase_firestore',
+          StandardMethodCodec(TestFirestoreMessageCodec()),
+        )..setMockMethodCallHandler((call) async {
+          return null;
+        });
+
+        final fooCollection = firestore.collection('foo').withConverter<int>(
+          fromFirestore: (ds, _) => 42,
+          toFirestore: (v, _) => {'key': 42},
+        );
+
+        expect(
+          firestore.collection('bar').doc().set({
+            'key': fooCollection.doc()
+          }),
+          completion(equals(null)),
+        );
```

Comment:
```suggestion
        await firestore.collection('bar').doc().set({'key': fooCollection.doc()});

        await expectLater(
          firestore.collection('bar').doc().get().then((s) => s.data,
          completion(equals(42)),
        );
```

Reply:
Unfortunately, this will not work. The test function is not asynchronous, so 'await' will fail. Also, reading if the result of a get is 42 seems misleading. We want to test if the dart code can correctly translate our reference during a write operation. So we want to check if the set operation runs successfully but not the following get operation.